src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -2213,6 +2223,7 @@ int sd_dhcp_client_new(sd_dhcp_client **ret, int anonymize) {
                 .anonymize = !!anonymize,
                 .max_attempts = UINT64_MAX,
                 .ip_service_type = -1,
+                .socket_priority = 0,
```

Comment:
all fields that aren't listed are zero-initialized anyway, hence please drop this line

Reply:
> Hence what I said earlier: this should be two fields internally: the priority integer _and_ a boolean that indicates whether to actually apply it.

Thanks, I added it in the very last updated commit, please let me know about it.
I also applied changes according to your other remarks.